bin/picca_deltas.py

```diff
@@ -353,68 +403,62 @@ def main():
+    #-- Mean continuum : mean_cont
+    log_lambda_temp = (Forest.log_lambda_min + np.arange(2) *
+                      (Forest.log_lambda_max - Forest.log_lambda_min))
+    log_lambda_rest_frame_temp = (Forest.log_lambda_min_rest_frame + np.arange(2) *
+                                  (Forest.log_lambda_max_rest_frame 
+                                   - Forest.log_lambda_min_rest_frame))
     Forest.get_var_lss = interp1d(
-        (Forest.log_lambda_min + np.arange(2) *
-         (Forest.log_lambda_max - Forest.log_lambda_min)),
+        log_lambda_temp,
         0.2 + np.zeros(2),
         fill_value="extrapolate",
         kind="nearest")
-    Forest.get_eta = interp1d((Forest.log_lambda_min + np.arange(2) *
-                               (Forest.log_lambda_max - Forest.log_lambda_min)),
-                              np.ones(2),
-                              fill_value="extrapolate",
-                              kind="nearest")
+    Forest.get_eta = interp1d(
+        log_lambda_temp,
+        np.ones(2),
+        fill_value="extrapolate",
+        kind="nearest")
     Forest.get_fudge = interp1d(
-        (Forest.log_lambda_min + np.arange(2) *
-         (Forest.log_lambda_max - Forest.log_lambda_min)),
+        log_lambda_temp,
         np.zeros(2),
         fill_value="extrapolate",
         kind="nearest")
     Forest.get_mean_cont = interp1d(
-        (Forest.log_lambda_min_rest_frame + np.arange(2) *
-         (Forest.log_lambda_max_rest_frame - Forest.log_lambda_min_rest_frame)),
+        log_lambda_rest_frame_temp, 
         1 + np.zeros(2))
-    # end of setup forest class variables
 
-    ### check that the order of the continuum fit is 0 (constant) or 1 (linear).
+    #-- Check that the order of the continuum fit is 0 (constant) or 1 (linear).
     if args.order:
         if (args.order != 0) and (args.order != 1):
             userprint(("ERROR : invalid value for order, must be eqal to 0 or"
                        "1. Here order = {:d}").format(args.order))
             sys.exit(12)
 
-    ### Correct multiplicative pipeline flux calibration
+    #-- Correct multiplicative pipeline flux calibration
     if args.flux_calib is not None:
-        try:
-            hdul = fitsio.FITS(args.flux_calib)
-            stack_log_lambda = hdul[1]['loglam'][:]
-            stack_delta = hdul[1]['stack'][:]
-            w = (stack_delta != 0.)
-            Forest.correct_flux = interp1d(stack_log_lambda[w],
-                                           stack_delta[w],
-                                           fill_value="extrapolate",
-                                           kind="nearest")
-            hdul.close()
-        except (OSError, ValueError):
-            userprint(("ERROR: Error while reading flux_calib"
-                       "file {}".format(args.flux_calib)))
-            sys.exit(1)
-
-    ### Correct multiplicative pipeline inverse variance calibration
+        hdul = fitsio.read(args.flux_calib, ext=1)
```

Comment:
why are we removing the try/except block here? I think it is useful to inform the user if an error is occuring here

Reply:
I removed because I was having the error message but it was not informative enough. 
Removing the try/except block I have more information on the error and the code stops anyway.

Reply:
Ok, I see